pyproject.toml

```
@@ -28,3 +14,80 @@ exclude = [
 
 # [tool.django-stubs]
 # django_settings_module = "dandiapi.settings"
+
+[tool.ruff]
+line-length = 100
+target-version = "py311"
+select = ["ALL"]
+ignore = [
+  # Incompatible with formatter
+  # https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules
+  "COM812", # missing-trailing-comma
+  "COM819", # prohibited-trailing-comma
+  "D206", # indent-with-spaces
+  "D300", # triple-single-quotes
+  "E111", # indentation-with-invalid-multiple
+  "E114", # indentation-with-invalid-multiple-comment
+  "E117", # over-indented
+  "ISC001", # single-line-implicit-string-concatenation
+  "ISC002", # multi-line-implicit-string-concatenation
+  "Q", # flake8-quotes
+  "W191", # tab-indentation
+
+  "A003", # Class attribute is shadowing a Python builtin
+  "ANN", # flake8-annotations
+  "ARG001", # Unused function argument
+  "ARG002", # Unused method argument
+  "D100", # Missing docstring in public module
+  "D101", # Missing docstring in public class
+  "D102", # Missing docstring in public method
+  "D103", # Missing docstring in public function
+  "D104", # Missing docstring in public package
+  "D105", # Missing docstring in magic method
+  "D106", # Missing docstring in public nested class
+  "D107", # Missing docstring in __init__
+  "EM101", # Exception must not use a string literal, assign to variable first
+  "EM102", # Exception must not use an f-string literal, assign to variable first
+  "ERA001", # Found commented-out code
+  "FIX", # flake8-fixme
+  "TD002", # Missing author in TODO
+  "TD003", # Missing issue link on the line following this TODO
+  "TRY003", # Avoid specifying long messages outside the exception class
+
+  # Try to fix upstream
+  "RUF012", #  Mutable class attributes should be annotated with `typing.ClassVar`
+]
```

```suggestion
  "PTH119", # TODO: re-enable this when it's fixed
]
```

@brianhelba we discussed this yesterday, and decided that a somewhat significant change like #1782 would be better to decouple from the Ruff work. Do you have any objections to just temporarily ignoring this rule so we can merge this PR?

---

Added in https://github.com/dandi/dandi-archive/pull/1784/commits/71225896f95576c9ebfdebd43d3c55b7f2eb734a